repos/system_upgrade/el7toel8/actors/ziplcheckmultiplerescueentries/libraries/ziplcheckmultiplerescueentries.py

```diff
@@ -0,0 +1,36 @@
+from leapp import reporting
+
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def inhibit_if_multiple_rescue_entries_present(boot_entries):
+    """
+    Inhibits the upgrade process if the given boot entries contain more than one rescue entry.
+
+    A boot entry is recognized as a rescue entry when its title contains the `rescue` substring.
+    """
+    # Keep the whole information about boot entries not just their count as
+    # we want to provide user with the details
+    rescue_entries = []
+    for boot_entry in boot_entries:
+        if 'rescue' in boot_entry.title.lower():
+            rescue_entries.append(boot_entry)
+
+    if len(rescue_entries) > 1:
+        # Prepare the list of available rescue entries for user
+        rescue_entries_text = ''
+        for rescue_entry in rescue_entries:
+            rescue_entries_text += '{0}{1}'.format(FMT_LIST_SEPARATOR, rescue_entry.title)
+        zipl_config_path = '/etc/zipl.conf'
+
+        reporting.create_report([
+            reporting.Title('Multiple rescue boot entries present in the bootloader configuration.'),
+            reporting.Summary(
+                'The zipl configuration file {0} contains multiple rescue boot entries:{1}'
+                .format(zipl_config_path, rescue_entries_text)
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.BOOT]),
+            reporting.Remediation(hint='Remove rescue boot entries from the configuration and leave just one.'),
```

Comment:
It would be good, if we could link to a KB (knowledge base) article where it is described how to remove an entry.

Reply:
We've discussed this internally - a new ticket documenting this issue has been created.